spec/rspec/matchers/built_in/match_with_captures_spec.rb

```diff
@@ -0,0 +1,41 @@
+RSpec.describe "expect(regex).to match(string).with_captures" do
+
+  it "does match a regex with a missing capture" do
+    expect(/(a)(b)?/).to match("a123a").with_captures("a", nil)
+  end
+
+  it "does not match a regex with an incorrect match" do
+    expect(/(a)/).not_to match("a123a").with_captures("b")
+  end
+
+  it "matches a regex without named captures" do
+    expect(/(a)/).to match("a123a").with_captures("a")
+  end
+
+  it "does not match a regex that doesn't match" do
+    expect(/(a)/).not_to match("123").with_captures("a")
+  end
+
+  it "does not match a regex that doesn't match without captures" do
+    expect(/(a)/).not_to match("123").with_captures()
+  end
+
+  if RUBY_VERSION != "1.8.7"
+    it "matches a regex with named captures" do
+      expect(Regexp.new("(?<num>123)")).to match("a123a").with_named_captures(:num => "123")
+    end
+    it "matches a regex with a nested matcher" do
```

Comment:
I know this is copy pasta from #848 but there should be an additional line break here.